components/core/src/clp/ffi/ir_stream/search/ErrorCode.cpp

```diff
@@ -17,6 +17,12 @@ auto ErrorCategory::name() const noexcept -> char const* {
 template <>
 auto ErrorCategory::message(ErrorCodeEnum error_enum) const -> std::string {
     switch (error_enum) {
+        case ErrorCodeEnum::EncodedTextAstDecodingFailure:
+            return "Failed to decode the given encoded text AST.";
+        case ErrorCodeEnum::LiteralTypeUnexpected:
+            return "An unexpected literal type is reached.";
```

Comment:
```suggestion
            return "Unexpected literal type.";
```

I think given the error code name, it's better to make the message more general.